tools/distrib/docgen/all_lang_docgen.sh

```diff
@@ -0,0 +1,117 @@
+#! /bin/bash
+# Copyright 2020 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# A script to automatically generate API references and push to GitHub.
+# This script covers Core/C++/ObjC/C#/PHP/Python. Due to lack of tooling for
+# Cython, Python document generation unfortunately needs to compile everything.
+# So, this script will take couple minutes to run.
+#
+# Generate and push:
+#
+#     tools/distrib/docgen/all_lang-docgen.sh YOUR_GITHUB_USERNAME
+#
+# Just generate:
+#
+#     tools/distrib/docgen/all_lang-docgen.sh
+#
+
+set -e
+
+# Find out the gRPC version and print it
+GRPC_VERSION="$(grep -m1 -Eo ' version: .*' build_handwritten.yaml | grep -Eo '[0-9].*')"
+echo "Generating documents for version ${GRPC_VERSION}..."
+
+# Specifies your GitHub user name or generates documents locally
+if [ $# -eq 0 ]; then
+    read -r -p "- Are you sure to generate documents without push to GitHub? [y/N] " response
+    if [[ "${response[0]}" =~ ^([yY][eE][sS]|[yY])$ ]]; then
+        GITHUB_USER=''
+    else
+        echo "Generation stopped"
+        exit 1
+    fi
+else
+    if [ $# -eq 1 ]; then
+        GITHUB_USER=$1
+    else
+        echo "Too many arguments!"
+        exit 1
+    fi
+fi
+
+# Exits on pending changes; please double check for unwanted code changes
+git diff --exit-code
+git submodule update --init --recursive
+
+# Changes to project root
+dir=$(dirname "${0}")
+cd "${dir}/../../.."
+
+# Clones the API reference GitHub Pages branch
+PAGES_PATH="/tmp/gh-pages"
+rm -rf "${PAGES_PATH}"
+git clone https://github.com/grpc/grpc -b gh-pages "${PAGES_PATH}"
+
+# Generates Core / C++ / ObjC / PHP documents
+rm -rf "${PAGES_PATH}/core" "${PAGES_PATH}/cpp" "${PAGES_PATH}/objc" "${PAGES_PATH}/php"
+docker run --rm -it \
```

Comment:
Since we didn't `set -x`, it might be nice to add some log statements before each docker build, so the user is aware of what's eating up the time. I haven't actually run the script though, so if the existing output is sufficient to let the user know what's going on, feel free to ignore this comment.

Reply:
Log printing added.

The output is quite verbose now. The good side is that it makes any failure trace-able, but the bad side is that the printed log about the Docker build might be overwhelmed by other log statements.

Reply:
If you think it's better without the logging, feel free to remove it. I just wanted to be cautious about silent scripts that inexplicably fail.

Reply:
The added log itself is better than nothing. People might be able to search it in the printed output. Let's keep it.